contracts/staking/Staking.sol

```diff
@@ -366,13 +354,17 @@ abstract contract Staking is StakingV4Storage, GraphUpgradeable, IStakingBase, M
         // Allocation identifier validation
         require(_allocationID != address(0), "!alloc");
 
-        // The contract caller must be an authorized asset holder
-        require(__assetHolders[msg.sender] == true, "!assetHolder");
-
         // Allocation must exist
         AllocationState allocState = _getAllocationState(_allocationID);
         require(allocState != AllocationState.Null, "!collect");
 
+        // If the query fees are zero, we don't want to revert
+        // but we also don't need to do anything, so just return
```

Comment:
This changes slightly the behavior of the function as we now are not emitting the `RebatesCollected` event when `_tokens == 0`. I can't think why this could be a problem, but wanted to make sure we consider it.